torch/csrc/jit/tensorexpr/types.h

```diff
@@ -31,6 +31,42 @@ enum class ScalarType : int8_t {
   NumOptions
 };
 
+static inline const char* toString(ScalarType t) {
+#define DEFINE_CASE(_, name) \
+  case ScalarType::name:     \
+    return #name;
+
+  switch (t) {
+    AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_AND_QINTS(DEFINE_CASE)
+    DEFINE_CASE(_, Undefined)
+    DEFINE_CASE(_, Handle)
+    DEFINE_CASE(_, Uninitialized)
+    DEFINE_CASE(_, None)
+    DEFINE_CASE(_, NumOptions)
+    default:
+      return "UNKNOWN_SCALAR";
+  }
+
+  assert(false);
+#undef DEFINE_CASE
+}
+
+static inline ScalarType toDtype(const std::string& s) {
```

Comment:
This name at least is wrong, there is a Dtype class as well as a ScalarType enum.